packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -127,16 +129,12 @@ export async function runHttpQuery(
     // debug. Therefore, we need to do some unnatural things, such
     // as use NODE_ENV to determine the debug settings
     return throwHttpGraphQLError(500, [e], {
-      debug: debugFromNodeEnv(process.env.NODE_ENV),
+      debug: debugFromNodeEnv(),
     });
   }
 
   if (options.debug === undefined) {
-    const nodeEnv =
-      '__testing_nodeEnv__' in options
-        ? options.__testing_nodeEnv__
-        : process.env.NODE_ENV;
-    options.debug = debugFromNodeEnv(nodeEnv);
+    options.debug = debugFromNodeEnv(options.nodeEnv);
```

Comment:
Here it seems to work. `options` is the result from `resolveGraphqlOptions`, which can theoretically set `nodeEnv`. In the case it sets it to something undefined, it should probably default to `NODE_ENV`.